### ISSUE TITLE
Deinterlacing Analyzer (Request)

### ISSUE THREAD

**Issue by jlw4049** [staxrip/staxrip#552]
I'd really like StaxRip to have the ability to 'Analyse' source for Deinterlacing. Sort of how MeGui does it. You can do it manually or it'll automatically detect it during batch encodes.

![image](https://user-images.githubusercontent.com/48299282/109378421-66d0ea00-78a0-11eb-9052-b9700240e9b6.png)

I know how to do this manually, but MeGui does check this for you automatically, I'm not sure how it works however it's something I'd eventually like added to StaxRip. 

Thanks for taking the time to read this! I appreciate the constant updates/work on this!

**Comment by Patman86** [comment]
[SourceDetector](https://github.com/Kurtnoise/megui/blob/6ce4f08da2f69d7a80555c85386c2fac1d49a682/core/gui/SourceDetector.cs) is used for this

**Comment by jlw4049** [comment]
I think it would be nice to add in!

**Comment by stax76** [comment]
I found stax comments in this file, so I googled and found:

https://forum.doom9.org/showthread.php?t=104760

But I'm not really remembering anything. 😟

**Comment by jlw4049** [comment]
It works quite well with MeGui but I wouldn't have the slightest idea how to implement it.

**Comment by stax76** [comment]
> You can do it manually or it'll automatically detect it during batch encodes.

For batch processing or anything automated I think it would be required having this code in an external console app like the autocrop tool:

https://github.com/staxrip/staxrip/tree/master/Tools/AutoCrop

In that case it would not be necessary to convert it to VB.NET, it would be required to convert the frame server code from VB.NET to C#, but that's easy since the dotpeek decompiler can be used.

It's probably much work.

**Comment by jlw4049** [comment]
> > You can do it manually or it'll automatically detect it during batch encodes.
> 
> For batch processing or anything automated I think it would be required having this code in an external console app like the autocrop tool:
> 
> https://github.com/staxrip/staxrip/tree/master/Tools/AutoCrop
> 
> In that case it would not be necessary to convert it to VB.NET, it would be required to convert the frame server code from VB.NET to C#, but that's easy since the dotpeek decompiler can be used.
> 
> It's probably much work.

If its to much I can live without it. It's a nice feature though.

**Comment by TheRealDarkVenom** [comment]
Yes please, this feature would be really nice and helpful, it's saves much of our encoding time

**Comment by Dendraspis** [comment]
@TheRealDarkVenom 

> [..] it's saves much of our encoding time

To be precise it would make it slower.  🤔

**Comment by 44vince44** [comment]
It would be helpful for someone to describe what this is supposed to do exactly...
As much as i understands, you are requesting a tool that checks, BEFORE ENCODING, the nature of the video by analyzing some frames.
Then what do we expect afterwards, once the tool knows what king of video ?
There are many interlaing situations:
- Telecine => apply IVTC
- PAL field order => swap fields
- combing problems due to bad deinterlacings, resizing without deinterlacing => **many** possible solutions depending the error in previous processing that leads to those symptoms

Is that what you want, a tools that does all those ? If so, don't hope for anything, this could not be automated. Only the first two cases can be automated.
And the first two cases can be also detected but doing a frame by frame visualization in the preview.
I don't think it is the role of Staxrip coders to provide with such a tool.
Analyze you videos, store them in folders according to their nature, then batch process every folder with proper parameters.
Staxrip helps users do their encodes, but users still have to strain their wit a bit .

**Comment by TheRealDarkVenom** [comment]
Well, in my case it would be really helpful some direction if my video needs some deinterlacing, this will be most used with 30 FPS videos who needs to crop or resize and when encodes the result of output video is that it has some weird "ghost lines" and to figure it oput I would need to spend some time to make tweeks manually or if I do it with a movie I would had to wait until encodes and see if everything is ok, to avoyd this the "deinterlacing analyzer" would detect the source and put the correect filter, I had to do that this week with a 30FPS movie to encode and make it to 24FPS, I used Megui to know the source (IVTC) an put the filter on Staxrip to make it right

**Comment by 44vince44** [comment]
@TheRealDarkVenom
Well this is exactly my point:
You need to detect telecine in your video, not just any interlacing. A telecine detection tool can be done, but not an "any interlacing" detection tool.

I feel no coder here is willing to do it...

Here is how you should do it.
Open your file in a serious player that allows frame/frame forward.
mpv.net is included inside Staxrip and does that.
Drop your file in mpv.net, place the cursor somewhere in the middle, and use the dot `.` on your keyboard to move 1 frame forward. If you count 3 frames ok then 2 interlaced, and this repeats all the time, then you have telecine.
Once you know that this video is telecine, move it to a folder where you'll put all your telecine videos.
Once you've grouped all the telecide videos together, open Staxrip, prepare your settings, and add filter Field>IVTC.
Then create a batch for all the files in this folder.

**Comment by mirh** [comment]
See also 
https://github.com/Selur/MPlayerInterlaceDetection
https://github.com/mpv-player/mpv/blob/master/TOOLS/idet.sh
